isis/tests/FunctionalTestsLronacpho.cpp

```diff
@@ -0,0 +1,134 @@
+    */
+    double algo3Result = 0.26584613;
+    
+    // dop! lineManager is not zero based
+    LineManager oLine(outputCube);
+    oLine.SetLine(300);
+    outputCube.read(oLine);
+ 
+    EXPECT_NEAR(oLine[300],algo3Result, 0.002);
+    outputCube.close();
+  }
+  catch(IException &e){
+    FAIL() << "Unable to open output cube: " <<e.toString().toStdString().c_str() << std::endl;
+  }
+}
+
+TEST(LronacphoNoDemAlgo2, FunctionalTestsLronacpho) {
+  QTemporaryDir prefix;
+
+  ASSERT_TRUE(prefix.isValid());
+
+  QString inCubeFileName = "data/lronacpho/M143947267L.cal.echo.crop.cub";
+  QString outCubeFileName = prefix.path() + "/output.cub";
+  QString phoParameterFileName = "data/lronacpho/NAC_PHO_LROC_Empirical.0002.pvl";
+  QVector<QString> args = {"from="+inCubeFileName, "to="+outCubeFileName, "PHOPAR="+phoParameterFileName, "usedem=no"};
+ 
+  UserInterface options(APP_XML, args);
+  try {
+    lronacpho(options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to apply photometry algorithm to LRO image: " <<e.toString().toStdString().c_str() << std::endl;
+  }
+  try{
+    // open output cube to test al algorithm was applied correctly
+    Cube outputCube;
+    outputCube.open(outCubeFileName, "r");
+   
+    double algo2Result = 0.28940132;
+
+    // dop! lineManager is not zero based
+    LineManager oLine(outputCube);
+    oLine.SetLine(300);
+    outputCube.read(oLine);
+ 
+    EXPECT_NEAR(oLine[300],algo2Result, 0.002);
+
+    outputCube.close();
+  }
+  catch(IException &e){
+    FAIL() << "Unable to open output cube: " <<e.toString().toStdString().c_str() << std::endl;
+  }
+}
+
+TEST(LronacphoNoDemAlgoDefault, FunctionalTestsLronacpho) {
+  QTemporaryDir prefix;
+
+  ASSERT_TRUE(prefix.isValid());
+
+  QString inCubeFileName = "data/lronacpho/M143947267L.cal.echo.crop.cub";
+  QString outCubeFileName = prefix.path() + "/output.cub";
+  QString phoParameterFileName = "data/lronacpho/NAC_PHO_LROC_Empirical.0003.pvl";
+  QVector<QString> args = {"from="+inCubeFileName, "to="+outCubeFileName, "usedem=no"};
+ 
+  UserInterface options(APP_XML, args);
+  try {
+    lronacpho(options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to apply photometry algorithm to LRO image: " <<e.toString().toStdString().c_str() << std::endl;
+  }
+  try{
+    // open output cube to test al algorithm was applied correctly
+    Cube outputCube;
+ 
+    outputCube.open(outCubeFileName, "r");
+    
+    double algo3Result = 0.26584613;
+
+    LineManager oLine(outputCube);
+    oLine.SetLine(300);
+    outputCube.read(oLine);
+ 
+    EXPECT_NEAR(oLine[300],algo3Result, 0.002);
+    outputCube.close();
```

Comment:
Anything particular about this DN that makes it important to check or is it a random sample?

Reply:
This is just a random sample compared to the same output using fx equation the science team is using manually. I would've included testing against fx so any random pixel value could be checked but fx isn't yet callable. This should be updated once it is in case a cube that doesn't have (300,300) pixel.